flutter-for-ios.md

```diff
@@ -27,3 +21,1847 @@ that are most relevant to your needs.
+community plugin.
+
+## How do I access the Camera?
+
+A popular plugin to access the camera is [`image_picker`](https://pub.dartlang.org/packages/image_picker).
+
+## How do I Log in with Facebook?
+
+To Log in with Facebook, you can use the
+[`flutter_facebook_login`](https://pub.dartlang.org/packages/flutter_facebook_login) community plugin.
+
+## How do I use Firebase features?
+
+Most Firebase functions are covered by [first or third party plugins](https://pub.dartlang.org/flutter/packages?q=firebase).
+For example:
+
+ * [`firebase_admob`](https://pub.dartlang.org/packages/firebase_admob) for Firebase AdMob
+ * [`firebase_analytics`](https://pub.dartlang.org/packages/firebase_analytics) for Firebase Analytics
+ * [`firebase_auth`](https://pub.dartlang.org/packages/firebase_auth) for Firebase Auth
+ * [`firebase_database`](https://pub.dartlang.org/packages/firebase_database) for Firebase RTDB
+ * [`firebase_storage`](https://pub.dartlang.org/packages/firebase_storage) for Firebase Cloud Storage
+ * [`firebase_messaging`](https://pub.dartlang.org/packages/firebase_messaging) for Firebase Messaging (FCM)
+ * [`flutter_firebase_ui`](https://pub.dartlang.org/packages/flutter_firebase_ui) for quick Firebase Auth integrations (Facebook, Google, Twitter and email)
+
+## How do I build my own custom native integrations?
+
+If there is platform-specific functionality that Flutter or its community
+Plugins are missing, then you can build your own following [this tutorial](https://flutter.io/developing-packages/).
+
+Flutter's plugin architecture, in a nutshell, is a lot like using an Event bus in
+Android: you fire off a message and let the receiver process and emit a result
+back to you. In this case, the receiver would be code running on the native side
+on Android or iOS.
+
+# Databases and local storage
+
+## How do I access `UserDefaults` in Flutter?
+
+In iOS, you can store a collection of key-value pairs using a property list,
+known as the `UserDefaults`.
+
+In Flutter, you can access the equivalent functionality by using the
+[Shared Preferences plugin](https://pub.dartlang.org/packages/shared_preferences).
+
+This plugin wraps the functionality of both `UserDefaults` and the Android
+equivalent, `SharedPreferences`.
+
+## What is the equivalent to CoreData in Flutter?
+In iOS, you can use CoreData to store structured data. This is simply a
+layer on top of an SQL database, making it easier to make queries that
+relate to your models.
+
+In Flutter, you can access this functionality by using the SQFlite plugin 
+[SQFlite](https://pub.dartlang.org/packages/sqflite).
+
+# Notifications
+
+## How do I setup Push Notifications?
+
+In iOS, you would need to register your app on the developer portal to allow
+push notifications. In Flutter, you can access this functionality by using the
+`Firebase_Messaging` plugin. You can find full instructions on how to get set
+up [here](https://pub.dartlang.org/packages/firebase_messaging).
```

Comment:
Never link on "here". There are many reasons, but foremost it is not accessible-compliant. 
https://developers.google.com/style/link-text   and 
https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/

Better: See the [firebase_messaging](https://pub.dartlang.org/packages/firebase_messaging) plugin for more information on using the Firebase Cloud Messaging API.